test/extensions/filters/http/aws_lambda/aws_lambda_transformer_test.cc

```diff
@@ -254,7 +254,7 @@ TEST_F(AWSLambdaTransformerTest, TestConfigureRequestTransformerSignatureNoBody)
 
 
   EXPECT_EQ(transformedxAmzDateHeader[0]->value().getStringView(), "20010909T014640Z");
-  EXPECT_EQ(transformedAuthorizationHeader[0]->value().getStringView(), "AWS4-HMAC-SHA256 Credential=access key/20010909/us-east-1/lambda/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-invocation-type;x-amz-log-type, Signature=6ac2a229351a686ff5a93deb46896f2128a6800a885ef5324c0c0685aa21d786");
+  EXPECT_EQ(transformedAuthorizationHeader[0]->value().getStringView(), "AWS4-HMAC-SHA256 Credential=access key/20010909/us-east-1/lambda/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-invocation-type;x-amz-log-type, Signature=561409e1250c56044b11a3d6eaed9f3d3d9467f3214dfec4786a65381bdab23e");
```

Comment:
Note to reviewers: I'm not sure why this hash has changed. I am actively looking into it now. My testing indicates that requests are being accepted by Amazon (I will link solo-projects e2e tests that validate this), so I don't think this is problematic, but I am currently confused by _why_ it is occurring

Reply:
So this happened because this test was actually hitting the bug that this PR resolves (should've known from the name `TestConfigureRequestTransformerSignatureNoBody`)

On master, running this tests confirms that the "transformed" data buffer used in this test is just an empty string. Now that that buffer contains the expected value, the hash of that value has changed.